### ISSUE TITLE
Progress Wiped

### ISSUE THREAD

**Issue by HealthyMindFrsh** [Gloryness/tassomai-automation#58]
My progress from the past 30 days seems to have been deleted, has the program been detected?

**Comment by Gloryness** [comment]
Yeah, I've already mentioned this but for more about it see #55
And that is why I added the "Lookup" tab, it's designed so you have to manually do the quiz but it will fetch the answers for you. That way, it won't wipe progress.

**Comment by Nuckerr** [comment]
Why is it wiping? Is there some sort of 'anti-bot'?

**Comment by realdeadbeef** [comment]
I can only assume it's detecting the fact that python is using selenium automation, could try using something like chromedriver or gekodriver

**Comment by realdeadbeef** [comment]
Okay, my bad. It doesn't use selenium, perhaps using API requests is easily detected as there would probably be other communications that aren't responded to. Using something like chromedriver might fix it.

**Comment by realdeadbeef** [comment]
I'm going to see how far i can get with selenium chromedriver

**Comment by Gloryness** [comment]
> 
> 
> Okay, my bad. It doesn't use selenium, perhaps using API requests is easily detected as there would probably be other communications that aren't responded to. Using something like chromedriver might fix it.

Yeah, when sending a post API request to answer a question it takes in 2 params: answer_id and content_id.
content_id is new and the program never adds in the param, only answer_id. Therefore this is maybe how its detecting the bot? since content_id is never given. (i do not know how they generate the value of content_id so its not like i can just add it in otherwise it'd be invalid)

**Comment by Gloryness** [comment]
> 
> 
> I'm going to see how far i can get with selenium chromedriver

tassomai automation used selenium a long time ago during the early stages of when it was released, but tassomai made it that selenium is detectable and plus it's more efficient using requests and user-friendly

**Comment by realdeadbeef** [comment]
I got it to open up a bot-controlled firefox window and log in but it knows i'm using selenium, not sure where to go from here

**Comment by realdeadbeef** [comment]
Perhaps some way of making selenium undetectable

**Comment by BIGboyal** [comment]
@realdeadbeef I'm working on a undetectable tassomai bot using this https://github.com/ultrafunkamsterdam/undetected-chromedriver , which is a selenium patch that is undetectable. I've done several quizzes using it so far that all haven't been wiped. but using selenium is a real pain for automation like this so it will take me a while to finish.

**Comment by realdeadbeef** [comment]
> @realdeadbeef I'm working on a undetectable tassomai bot using this https://github.com/ultrafunkamsterdam/undetected-chromedriver , which is a selenium patch that is undetectable. I've done several quizzes using it so far that all haven't been wiped. but using selenium is a real pain for automation like this so it will take me a while to finish. 

Care to share? Perhaps we could talk about it on discord?

**Comment by Nuckerr** [comment]
Pop it on a new branch

**Comment by HealthyMindFrsh** [comment]
Think it's possible to extract answers from the progress tree? Now that the automation has been detected there isn't really a way to add new questions to the database now.

**Comment by Gloryness** [comment]
Yeah it's possible and surely implementable as from what I can see and will definitely help. At the current moment the automation will still work, Tassomai just picks up on it and then wipes progress. This does not mean that it can't find out whether the question was correct or not so it is still able to add questions to the database.

**Comment by HealthyMindFrsh** [comment]
so after like a year I finally got caught lmao, apparently it's because my reaction times were 0.1 seconds, maybe it has something to do with that?

**Comment by Gloryness** [comment]
It has 226k lines of code lmao you can find it here
![Screenshot 2021-10-14 16 47 00](https://user-images.githubusercontent.com/64639943/137352047-1d4d0acf-5cbe-4da0-a60c-be33cf9cf757.png)

**Comment by w3bhook** [comment]
https://github.com/Gloryness/tassomai-automation/pull/63

**Comment by Nuckerr** [comment]
Tassomai have their eyes on this repo. Def no way to continue this without making it closed source :(

**Comment by w3bhook** [comment]
we'll try some stricter methods of securing our patches next time xd

**Comment by SkyloKobe** [comment]
Hey, they have started to remove points and progress again. Is this normal after the fix?

**Comment by w3bhook** [comment]
yes it is normal, the fix worked for about 2 weeks, but eventually tassomai caught on and patched it. I didn't bother to send an update message here since the issue was reopened by Gloryness.

**Comment by realdeadbeef** [comment]
Is this still a problem?
